src/hooks/useLoadDisciplines.ts

```diff
@@ -23,19 +22,20 @@ const getEndpoint = (parent: Discipline | null): string => {
   }
 }
 
-const formatServerResponse = (serverResponse: ServerResponse[], apiKey?: string): DisciplineType[] =>
+const formatServerResponse = (serverResponse: ServerResponse[], parent: DisciplineType | null): DisciplineType[] =>
   serverResponse.map(item => ({
     ...item,
     numberOfChildren: item.total_discipline_children || item.total_training_sets || item.total_documents,
-    isLeaf: item.total_discipline_children === 0,
-    apiKey: apiKey
+    isLeaf: item.total_documents !== undefined,
```

Comment:
Why?

Reply:
Because the previous check did not check if it is a leaf :)